libs/init/buildgen/src/mill/main/buildgen/BuildGenBase.scala

```diff
@@ -4,13 +4,12 @@ import mill.main.buildgen.BuildGenUtil.{compactBuildTree, writeBuildObject}
 
 import scala.collection.immutable.{SortedMap, SortedSet}
 
-/*
-TODO Can we just convert all generic type parameters to abstract type members?
- See https://stackoverflow.com/a/1154727/5082913.
- I think abstract type members are preferred in this case.
- */
-trait BuildGenBase[M, D, I] {
+trait BuildGenBase {
+  type M
+  type D
+  type I
   type C
```

Comment:
Can we have more descriptive names here?

Reply:
Yeah, they are respectively `Module`, `Dependency`, `Input`, and `Config`.

Reply:
I also just noticed that `D`/`Dependency` is no longer used in the current implementation. Shall I remove it or just keep it for possible use and better readability?

Reply:
If it's unused, remove it.